clouditor-engine-core/src/main/java/io/clouditor/assurance/Rule.java

```diff
@@ -32,23 +32,63 @@
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import io.clouditor.assurance.ccl.*;
 import io.clouditor.discovery.Asset;
+import java.io.Serializable;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
+import javax.persistence.*;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.hibernate.annotations.LazyCollection;
+import org.hibernate.annotations.LazyCollectionOption;
 
-public class Rule {
+@Entity(name = "rule")
+@Table(name = "rule")
+public class Rule implements Serializable {
+
+  private static final long serialVersionUID = -5934273783785749037L;
 
   @JsonDeserialize(contentUsing = CCLDeserializer.class)
   @JsonSerialize(contentUsing = CCLSerializer.class)
+  @ManyToMany
+  @LazyCollection(LazyCollectionOption.FALSE)
+  @JoinTable(
+      name = "condition_to_many_rules",
+      joinColumns = @JoinColumn(name = "rule_id", referencedColumnName = "rule_id"),
+      inverseJoinColumns = {
+        @JoinColumn(name = "source", referencedColumnName = "source"),
+        @JoinColumn(name = "type_value", referencedColumnName = "type_value"),
+      })
   private List<Condition> conditions = new ArrayList<>();
 
+  @Column(name = "active")
   private boolean active;
 
-  @JsonProperty private String name;
-  @JsonProperty private String description;
-  @JsonProperty private String icon = "far fa-file-alt";
-  @JsonProperty private List<String> controls = new ArrayList<>();
-  @JsonProperty private String id;
+  @JsonProperty
+  @Column(name = "rule_name")
+  private String name;
+
+  @JsonProperty
+  @Column(name = "rule_description")
+  private String description;
+
+  @JsonProperty
+  @Column(name = "icon")
+  private final String icon = "far fa-file-alt";
+
+  @JsonProperty
+  @ManyToMany
+  @LazyCollection(LazyCollectionOption.FALSE)
+  @JoinTable(
+      name = "rule_to_control",
+      joinColumns = @JoinColumn(name = "rule_id", referencedColumnName = "rule_id"),
+      inverseJoinColumns = @JoinColumn(name = "control_id", referencedColumnName = "control_id"))
+  private final List<Control> controls = new ArrayList<>();
```

Comment:
uh, changing this might have implications on the UI. did you check if that is still working? the UI probably expects a list of strings here.